### DISCUSSION TITLE
How to mock a whole store?

### DISCUSSION THREAD

**Discussion by jaytonic** [zuriscript/signalstory#97] [Q&A] [answered]
Hi,

I'm unit-testing one service, that receives a store in the constructor.
I'm trying to avoid having to inject the real store, do some commands, ... to only test the service.

But I'm struggling to mock signal or computed signals, did someone already tried this?

I'm using jest and jest-auto-spies.

Thanks!

**Comment by zuriscript** [comment, marked as answer]
Hi @jaytonic 

Thanks a lot for starting this discussion.
Sidenote: In the docs I have covered how to test a [store in isolation](https://zuriscript.github.io/signalstory/docs/testing).
This would also include testing for advanced flows like signal changed value propagation etc. which is well covered in signalstory own [test suite](https://github.com/zuriscript/signalstory/tree/master/packages/signalstory/src/__tests__).

When mocking a store with exposed signals to the outside world, I would mock the signal return value, but otherwise let it stay a signal to conserve the inner mechanics. I guess, especially including getter semantics,  this makes it a bit more involved to configure the testing libraries accordingly.

I have answered your [specific question on stackoverflow](https://stackoverflow.com/questions/78968089/how-to-mock-an-angular-class-with-signals). I don't have much experience with `jest-auto-spies`, though.

Essentially, there are many ways to do it and I might do a blog post in the future, or cover it in the documentation, as soon as time permits.

**Reply by jaytonic** [reply]
Hey @zuriscript !
Thanks for the help! I did read and follow your documentation(very useful!)
But in my case, it was more about testing some service that does depends on the store, which was not covered by the test.

I came to the same conclusion as you by "chance" with the method instead of properties, it works perfectly, I've a great coverage on both store and service using store.

Ps: We started to use your great library in our solution that is meant to evolve on the next 10 years, quite happy with it for now :)
